base/compiler/typelimits.jl

```diff
@@ -261,7 +261,8 @@ function type_more_complex(@nospecialize(t), @nospecialize(c), sources::SimpleVe
         elseif isa(c, DataType) && t.name === c.name
             cP = c.parameters
             length(cP) < length(tP) && return true
-            length(cP) > length(tP) && !isvarargtype(tP[end]) && depth == 1 && return false
+            length(cP) > length(tP) && isvarargtype(cP[end]) && return false
```

Comment:
I might have over-complicated the example on you, since the error will occur even if cP does not have a Vararg:
```
julia> Core.Compiler.type_more_complex(Tuple{Vararg{Tuple{}}}, Tuple{Tuple{Int,Int}}, Core.svec(), 0, 0, 0)
ERROR: BoundsError: attempt to access Core.SimpleVector at index [0]
```

I think what we need is to check again (after the unwrapva call earlier) that there are some parameters here to check:
```suggestion
             isempty(tP) && return false
```

Reply:
You are right. And Vararg is more complex Int,Int. I rethink this function's purpose. thank you.

Reply:
I have an question in this situation,
```
julia> Core.Compiler.type_more_complex(Tuple{Int,Int}, Tuple{Vararg{Int}}, Core.svec(), 1, 1, 1)
true
``
is it expected a result? I think `Tuple{Int, Int}` is not more complex than `Tuple{Vararg{Int}}` because It is not .

Reply:
I survey this code blame and rethink it and agree your suggestion.